app/command_groupmsg.go

```diff
@@ -0,0 +1,154 @@
+	}
+
+	matches := messageRegex.FindStringSubmatch(message)
+
+	for _, username := range strings.Split(matches[1], ",") {
+		username = strings.TrimSpace(username)
+		username = strings.TrimPrefix(username, "@")
+		if result := <-a.Srv.Store.User().GetByUsername(username); result.Err != nil {
+			invalidUsernames = append(invalidUsernames, username)
+		} else {
+			targetUser := result.Data.(*model.User)
+			_, exists := targetUsers[targetUser.Id]
+			if !exists && targetUser.Id != args.UserId {
+				targetUsers[targetUser.Id] = targetUser
+				targetUsersSlice = append(targetUsersSlice, targetUser.Id)
+			}
+		}
+	}
+
+	if len(invalidUsernames) == 1 {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.invalid_user.app_error"), "@"+invalidUsernames[0]),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+	if len(invalidUsernames) > 1 {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.invalid_users.app_error"), "@"+strings.Join(invalidUsernames, ", @")),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	if len(matches) > 2 {
+		parsedMessage = matches[2]
+	}
+
+	if len(targetUsersSlice) == 2 {
+		return GetCommandProvider("msg").DoCommand(a, args, fmt.Sprintf("%s %s", targetUsers[targetUsersSlice[1]].Username, parsedMessage))
+	}
+
+	if len(targetUsersSlice) < model.CHANNEL_GROUP_MIN_USERS {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.min_users.app_error"), model.CHANNEL_GROUP_MIN_USERS-1),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	if len(targetUsersSlice) > model.CHANNEL_GROUP_MAX_USERS {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.max_users.app_error"), model.CHANNEL_GROUP_MAX_USERS-1),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	// Find the channel based on this user
+	channelName := model.GetGroupNameFromUserIds(targetUsersSlice)
+
+	targetChannelId := ""
+	if channel := <-a.Srv.Store.Channel().GetByName(args.TeamId, channelName, true); channel.Err != nil {
+		if channel.Err.Id == "store.sql_channel.get_by_name.missing.app_error" {
+			if groupChannel, err := a.CreateGroupChannel(targetUsersSlice, args.UserId); err != nil {
+				l4g.Error(err.Error())
+				return &model.CommandResponse{Text: args.T("api.command_groupmsg.group_fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+			} else {
+				targetChannelId = groupChannel.Id
+			}
+		} else {
+			l4g.Error(channel.Err.Error())
+			return &model.CommandResponse{Text: args.T("api.command_groupmsg.group_fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+		}
+	} else {
+		channel := channel.Data.(*model.Channel)
+		targetChannelId = channel.Id
+		teamId = channel.TeamId
```

Comment:
All group message channels have an empty teamId, so this will always be empty. That'll simplify the logic below a bit since you won't need to check the session, and you can just use `args.TeamId` when getting the team name to construct the URL.

That'll also prevent it from changing teams since I think it may do that right now when the current team doesn't match up with the one for `args.Session.TeamMembers[0]`